solo-io/gloo
# Fix destinationOverrides not being applied to empty aws destination

# Description

This fixes an issue with `destinationOverrides` not being set if the destination was unset for aws routes.
Adds tests to check the outroute has the applied overrides, and moves tests with "no destinationSpec" to own context.

# Context

While testing more upstream & virtual service routing edge cases I fell upon this.

# Testing

1. Install Gloo with custom `values.yaml` to disable lambda discovery & with fallbackToFirstFunction enabled: `discovery.fdsMode: DISABLED; settings.aws.fallbackToFirstFunction: true`
2. Create an AWS secret 
3. Create an Upstream with a lambda function manually added to it and a `destinationOverride` set (ex. `destinationOverrides.requestTransformation: true`)
4. Create a Virtual Service pointing to upstream, **without** a `destinationSpec set`
5. Port-forward gloo-proxy to 8080
6. `curl -X POST -d '{"foo": "hello world"}' localhost:8080/`

Without this fix you'll see the request go through, but the `destinationOverride` will not take effect.
With this fix, you'll see the request go through as well as the `destinationOverride` taking effect.

# Checklist:
